javalib/src/main/scala/java/util/logging/Logger.scala

```diff
@@ -0,0 +1,252 @@
+  //def getAnonymousLogger(resourceBundle: String):Logger
+}
+
+class Logger private (protected val name: String,
+    protected val resourceBundle: Option[String],
+    protected var level: Option[Level],
+    protected var useParentsHandlers: Boolean,
+    protected var parent: Option[Logger] = None,
+    protected var filter: Option[Filter] = None) {
+
+  private[this] var handlers: Array[Handler] = Array.empty
+
+  protected def this(name: String, resourceBundle: String) =
+    this(name, Option(resourceBundle), None, true)
+
+  // Not implemented, no resource bundle
+  //def getResourceBundle():ResourceBundle = ???
+
+  // Find the effective level
+  private def levelR: Option[Level] = {
+    @tailrec
+    def go(logger: Logger): Option[Level] = logger.level match {
+      case l @ Some(_)                 => l
+      case None if getParent() != null => go(logger.getParent)
+      case None                        => None
+    }
+
+    go(this)
+  }
+
+  def getResourceBundleName():String = resourceBundle.orNull
+
+  def setFilter(filter: Filter): Unit = this.filter = Option(filter)
+
+  def getFilter(): Filter = filter.orNull
+
+  def log(record: LogRecord): Unit = if (isLoggable(record.getLevel)) {
+    if (useParentsHandlers) {
+      Logger.findParent(this).orElse(parent).foreach(_.log(record))
+    }
+    handlers.foreach(_.publish(record))
+  }
+
+  def log(level: Level, msg: String): Unit =
+    log(new LogRecord(level, msg, loggerName = Option(name)))
+
+  def log(level: Level, msg: String, param: AnyRef): Unit =
+    log(new LogRecord(level, msg, params = List(param),
+      loggerName = Option(name)))
+
+  def log(level: Level, msg: String, params: Array[AnyRef]): Unit =
+    log(new LogRecord(level, msg, params = params.toList,
+      loggerName = Option(name)))
+
+  def log(level: Level, msg: String, thrown: Throwable): Unit =
+    log(new LogRecord(level, msg, thrown = Option(thrown),
+      loggerName = Option(name)))
+
+  def logp(level: Level, sourceClass: String, sourceMethod: String,
+      msg: String): Unit =
+    log(new LogRecord(level, msg,
+      sourceClassName = Option(sourceClass),
+      sourceMethodName = Option(sourceMethod),
+      loggerName = Option(name)))
+
+  def logp(level: Level, sourceClass: String, sourceMethod: String,
+      msg: String, param: AnyRef): Unit =
+    log(new LogRecord(level, msg,
+      sourceClassName = Option(sourceClass),
+      sourceMethodName = Option(sourceMethod),
+      params = List(param),
+      loggerName = Option(name)))
+
+  def logp(level: Level, sourceClass: String, sourceMethod: String,
+      msg: String, params: Array[AnyRef]): Unit =
+    log(new LogRecord(level, msg,
+      sourceClassName = Option(sourceClass),
+      sourceMethodName = Option(sourceMethod),
+      params = params.toList,
+      loggerName = Option(name)))
+
+  def logp(level: Level, sourceClass: String, sourceMethod: String,
+      msg: String, thrown: Throwable): Unit =
```

Comment:
Add `{ }` to all the methods that have a body of 2 lines or more (exception with `if/else` expressions).